src/app/beer_garden/db/mongo/models.py

```diff
@@ -312,6 +313,7 @@ class Request(MongoModel, Document):
             {"name": "command_index", "fields": ["command"]},
             {"name": "command_type_index", "fields": ["command_type"]},
             {"name": "system_index", "fields": ["system"]},
+            {"name": "expiration_date_index", "fields": ["expiration_date"]},
```

Comment:
for optimization, you should have an index of both `expiration_date` and `status` and `has_parent` so we don't have to inspect every file while pruning